src/Microsoft.TestPlatform.CrossPlatEngine/Resources/Resources.resx

```diff
@@ -169,7 +169,8 @@
     <value>Logging TestHost Diagnostics in file: {0}</value>
   </data>
   <data name="TestHostExitedWithError" xml:space="preserve">
-    <value>Testhost process exited with error: {0}. Please check the diagnostic logs for more information.</value>
+    <value>Testhost process for source {0} exited with error: {1}. Please check the diagnostic logs for more information.</value>
```

Comment:
If it can be a list of sources, could we have a message that reflects it? Currently we would display something like `Testhost process for source a.cs, b.cs exited with error...`. As you say it's pretty rare, maybe we don't need multiple messages.

What about writing `for source(s)`? Maybe we could even add quotes like `for source(s) '{0}'` and in calling sites, instead of doing `string.Join(", ", sources)` you would do `string.Join("', '", sources)`.

Reply:
done